go/badges/badgestate.go

```diff
@@ -266,6 +277,24 @@ func (b *BadgeState) UpdateWithGregor(ctx context.Context, gstate gregor.State)
 				}
 				b.state.NewTeamNames = append(b.state.NewTeamNames, x.TeamName)
 			}
+		case "team.delete":
+			var body []teamDeletedBody
+			if err := json.Unmarshal(item.Body().Bytes(), &body); err != nil {
+				b.log.CDebugf(ctx, "BadgeState unmarshal error for team.delete item: %v", err)
+				continue
+			}
+			for _, x := range body {
+				if x.TeamName == "" || x.OpBy.Username == "" {
```

Comment:
@cjb should we also filter if the when `x.OpBy.Username == currentUsername`? what do the designs call for?